src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -286,7 +306,7 @@ async Task<bool> InnerProcessMessage(Dictionary<string, string> headers, string
                     var messageContext = new MessageContext(
                         nativeMessageId,
                         new Dictionary<string, string>(headers),
-                        body,
+                        body.ToArray(),
```

Comment:
This should not be necessary and actually invalidates the whole buffer pooling. The pool returns a byte array and the code that targets v8 just treats it `AsMemory`. In order to still benefit from the pooling approach, can simplify the code a bit for v7 to only use the `byte[]` reference.

Reply:
I have changed it to use bye[] in general - as it was doing originally.

Reply:
Awesome